src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+        $rule_info{error_msg} = "rule parser internal error (missing argument)";
+        return \%rule_info;
+    }
+    unless (defined($parser_options)) {
+        *$self->{LOG}->debug(2, "$function_name: 'parser_options' undefined");
+        $parser_options = {};
+    }
+    if (defined($parser_options->{always_rules_only})) {
+        *$self->{LOG}->debug(1, "$function_name: 'always_rules_only' option set to " . $parser_options->{always_rules_only});
+    } else {
+        *$self->{LOG}->debug(1, "$function_name: 'always_rules_only' option not defined: assuming $LINE_OPT_DEF_ALWAYS_RULES_ONLY");
+        $parser_options->{always_rules_only} = $LINE_OPT_DEF_ALWAYS_RULES_ONLY;
+    }
+
+    (my $condition, my $tmp) = split /->/, $rule;
+    if ($tmp) {
+        $rule = $tmp;
+    } else {
+        $condition = "";
+    }
+    *$self->{LOG}->debug(1, "$function_name: condition=>>>$condition<<<, rule=>>>$rule<<<");
+
+    # Check if only rules with ALWAYS condition must be applied.
+    # ALWAYS is a special condition that is used to flag the only rules that
+    # must be applied if the option always_rules_only is set. When this option
+    # is not set, this condition has no effect and is just reset to an empty conditions.
+    if ($parser_options->{always_rules_only}) {
+        if ($condition ne $RULE_CONDITION_ALWAYS) {
+            *$self->{LOG}->debug(1, "$function_name: rule ignored ($RULE_CONDITION_ALWAYS condition not set)");
+            return;
+        }
+    }
+    if ($condition eq $RULE_CONDITION_ALWAYS) {
+        $condition = '';
+    }
+
+    # Check if rule condition is met if one is defined
+    if ($condition ne "") {
+        *$self->{LOG}->debug(1, "$function_name: checking condition >>>$condition<<<");
+
+        # Condition may be negated if it starts with a !: remove it from the condition value.
+        # If the condition is negated, when the condition is true the rule must not be applied.
+        my $negate = 0;
+        if ($condition =~ /^!/) {
+            $negate = 1;
+            $condition =~ s/^!//;
+        }
+        my ($cond_attribute, $cond_option_set) = split /:/, $condition;
+        unless ($cond_option_set) {
+            $cond_option_set = $cond_attribute;
+            $cond_attribute  = "";
+        }
+        *$self->{LOG}->debug(2, "$function_name: condition option set = '$cond_option_set', "
+                             . "condition attribute = '$cond_attribute', negate=$negate");
+        my $cond_satisfied = 1;    # Assume condition is satisfied
+        if ($cond_attribute) {
+         # Due to Perl autovivification, testing directly exists($config_options->{$cond_option_set}->{$cond_attribute}) will spring
+         # $config_options->{$cond_option_set} into existence if it doesn't exist.
+            my $cond_true = $config_options->{$cond_option_set}
+              && exists($config_options->{$cond_option_set}->{$cond_attribute});
+            if ($negate) {
```

Comment:
Except that it doesn't work... no idea why. Keeping the original code for the time being.